model/model/v1alpha/model.proto

```diff
@@ -1737,3 +1738,117 @@ message UndeployOrganizationModelAdminRequest {
+    // model trigger in progress
+    TRIGGER_STATUS_PROCESSING = 1;
+    // model trigger succeeded
+    TRIGGER_STATUS_COMPLETED = 2;
+    // model trigger failed
+    TRIGGER_STATUS_FAILED = 3;
+    // model trigger is waiting to be executed
+    TRIGGER_STATUS_QUEUED = 4;
+  }
+
+  // TriggerSource defines the source of a model trigger
+  enum TriggerSource {
+    // Unspecified
+    TRIGGER_SOURCE_UNSPECIFIED = 0;
+    // model trigger from frontend UI
+    TRIGGER_SOURCE_CONSOLE = 1;
+    // model trigger from API or SDK
+    TRIGGER_SOURCE_API = 2;
+  }
+
+  // Model Trigger UUID.
+  string uid = 1 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Model UUID.
+  string model_uid = 2 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Pipeline Trigger UUID.
+  string trigger_uid = 3 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Model trigger status
+  TriggerStatus status = 4 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Model visibility.
+  Model.Visibility visibility = 5 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Trigger source
+  TriggerSource source = 6 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Trigger start time
+  google.protobuf.Timestamp start_time = 7 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Trigger total duration in milliseconds
+  optional google.protobuf.Duration total_duration = 8 [
+    (google.api.field_behavior) = OUTPUT_ONLY,
+    (google.api.field_behavior) = OPTIONAL
+  ];
+  // Trigger end time
+  optional google.protobuf.Timestamp end_time = 9 [
+    (google.api.field_behavior) = OUTPUT_ONLY,
+    (google.api.field_behavior) = OPTIONAL
+  ];
+  // Trigger requester UID.
+  string requester_uid = 10 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // The amount of Instill Credit consumed by the trigger. This field will only be present on Instill Cloud.
+  optional float credit_amount = 11 [
+    (google.api.field_behavior) = OUTPUT_ONLY,
+    (google.api.field_behavior) = OPTIONAL
+  ];
+  // Error message occurred during model trigger
+  optional string error = 12 [
+    (google.api.field_behavior) = OUTPUT_ONLY,
+    (google.api.field_behavior) = OPTIONAL
+  ];
+  // Model trigger creation time.
+  google.protobuf.Timestamp create_time = 13 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Model trigger update time.
+  google.protobuf.Timestamp update_time = 14 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // The model version identifier, which is equal to image tag.
+  string version = 15 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Tags.
+  repeated string tags = 16 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Task type.
+  common.task.v1alpha.Task task = 17 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Inference input parameters.
+  optional google.protobuf.Struct task_inputs = 18 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Model inference outputs.
+  repeated TaskOutput task_outputs = 19 [(google.api.field_behavior) = OUTPUT_ONLY];
```

Comment:
@joremysh since there wont be any release in sprint 41 for model-backend, I will be merging my PR for `protobuf` and `model-backend` soon. and both `task_inputs` and `task_outputs` will be an array of `google.protobuf.Struct`